### PR TITLE
Update documentation and code to remove reference to rate-checker-fees endpoint and Fee model

### PR DESCRIPTION
**Pull request by cwdavies**
The following updates were requested or are necessary as a results of removing Fee from Python code
1. update API and README documentation
2. remove rate-checker-fees endpoint URL
3. remove rate_checker_fees from view
4. add ratechecker migration to remove_fee_loader
5. add fix_fee_product_index to remove_fee_loader migration
6. remove fees from test_views Python code

### CODE REVIEW

**Comment by cwdavies** [comment]
Verified 0002 migration runs as follows:
$ python cfgov/manage.py migrate
Operations to perform:
  Apply all migrations: admin, agreements, ask_cfpb, auth, comparisontool, contenttypes, countylimits, data_research, flags, hmda, jobmanager, paying_for_college, prepaid_agreements, ratechecker, regulations3k, retirement_api, sessions, taggit, teachers_digital_platform, v1, wagtailadmin, wagtailcore, wagtaildocs, wagtailembeds, wagtailforms, wagtailimages, wagtailinventory, wagtailredirects, wagtailsharing, wagtailusers
Running migrations:
  Applying ratechecker.0002_remove_fee_loader... OK

File: ratechecker/migrations/0002_remove_fee_loader.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.22 on 2019-10-31 16:33
+from __future__ import unicode_literals
+
+from django.db import migrations, OperationalError, ProgrammingError
+
+def fix_fee_product_index(apps, schema_editor):
+    try:
+        schema_editor.execute(
+            'DROP INDEX IF EXISTS idx_16977_product_id;'
```
Comment (chosak) [inline ratechecker/migrations/0002_remove_fee_loader.py]: I don't think we can assume the constraint will be named `idx_16977_product_id`; isn't that name generated dynamically? Why is this needed? Won't the constraint automatically be deleted when the table is?

Comment (cwdavies) [reply ratechecker/migrations/0002_remove_fee_loader.py]: This constraint was not automatically deleted when the Fee table was deleted which is why a fix to the fee product index is needed and was recommended by @willbarton. When running the migrate without using this fix in 0002_remove_fee_loader.py the following error was logged: django.db.utils.ProgrammingError: constraint "idx_16977_product_id" of relation "ratechecker_fee" does not exist. I do not think that the constraint name is generated dynamically as it is defined in the production database dumpfile: production_django.sql.gz. I removed the lingering reference to the fees view in test_views.py and added that file to this PR. Here is the output from running tox on the command line:
```script
System check identified no issues (0 silenced).
......................................................................................................................................................................................................
----------------------------------------------------------------------
Ran 198 tests in 6.752s

OK
Destroying test database for alias 'default'...
py36-dj111 run-test: commands[2] | coverage report --skip-covered
Name                     Stmts   Miss  Cover
--------------------------------------------
ratechecker/dataset.py      53      1    98%
--------------------------------------------
TOTAL                     1855      1    99%

36 files skipped due to complete coverage.
______________________________________________________ summary ______________________________________________________
  py27-dj111: commands succeeded
  py36-dj111: commands succeeded
  congratulations :)
```

**Comment by cwdavies** [comment]
This is a valid concern from @chosak that the constraint has been hard-coded to idx_16977_product_id in the migration code and this name maybe dynamically generated. Please feel free to add your comments on whether the Python code should be changed to SELECT indexname FROM pg_indexes WHERE tablename='ratechecker_fee'. Thanks.

**Comment by willbarton** [comment]
We talked about this while pairing... and I think that because it occurs in our database and seems to [be the result of the MySQL -> PostgreSQL transition](https://stackoverflow.com/questions/47585826/django-unable-to-migrate-postgresql-constraint-x-of-relation-y-does-not-exist), it makes sense to hard-code the value we're having difficult with, so as to catch issues that exist in the database we're carrying around between servers/local development that theoretically should not exist in a way that Django can't manage otherwise.
